edit/node.vue

```diff
@@ -0,0 +1,60 @@
+<script>
+import LabeledInput from '@/components/form/LabeledInput';
+import Labels from '@/components/form/Labels';
+import Taints from '@/components/form/Taints';
+import Footer from '@/components/form/Footer';
+import CreateEditView from '@/mixins/create-edit-view';
+import { DESCRIPTION, HOSTNAME } from '@/config/labels-annotations';
+
+export default {
+  name: 'EditNode',
+
+  components: {
+    Footer, LabeledInput, Labels, Taints
+  },
+
+  mixins:     [CreateEditView],
+
+  props: {
+    value: {
+      type:     Object,
+      required: true,
+    },
+  },
+
+  data() {
+    return {
+      DESCRIPTION,
+      HOSTNAME
+    };
+  },
+};
+</script>
+
+<template>
+  <div class="node">
+    <div class="row">
+      <div class="col span-6">
+        <LabeledInput v-model="value.metadata.labels[HOSTNAME]" :mode="mode" type="text" label="Hostname" placeholder="e.g. storage" />
+      </div>
+    </div>
+    <div class="row">
+      <div class="col span-6">
+        <LabeledInput v-model="value.spec.displayName" :mode="mode" type="text" label="Custom Name" placeholder="e.g. ssd-storage" />
```

Comment:
This is a Rancher-ism for 2.x nodes (in the management plane), it doesn't exist in the native k8s node resource and won't actually be saved. (can't add arbitrary fields to the spec, API drops them).

I think you just want to show the standard `metadata.name` (which isn't editable either), and the normal description annotation.  Most of the stuff we talked about offline with requestedHostname, the actual hostname, the actual name, displayName, etc applies to the management-level nodes in Rancher, not the native node resource in k8s which is all you're dealing with here (for now).

Also consider using `<NameNsDescription>` instead of rolling your own.  It does extra stuff like focusing the name field, handles cases where `metadata.(labels|annotations)` are not defined on the object, and has a slot that could be used to make it show a different field than name, or add additional fields to the row.

Reply:
Ill go ahead and switch it over to NameNSDescription.

This was just following the design I had https://xd.adobe.com/view/4064e2e5-6616-4284-7081-c99e9314f431-3655/screen/900d417c-d741-4bc9-a013-b2f8223e4a1b/edit-node. But based on the new outline tani put together I should switch it over.